bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -542,7 +546,7 @@ impl Connection {
     // Poll the connection future if it exists.
     //
     // If the future returns Pending, then re-set it back on the Connection.
-    fn poll_async_task(&mut self) -> Option<Poll<Result<(), Error>>> {
+    pub(crate) fn poll_async_task(&mut self) -> Option<Poll<Result<(), Error>>> {
```

Comment:
I really don't want renegotiate polluting our main modules. The average developer should never know renegotiate exists. Also, while the renegotiate source code was only like ~200 lines, I had ~500 lines of testing. connection.rs and config.rs are already very large.

But I'm not a fan of all the random pub(crate)s either. I've done a bit of a refactor, see if you like this better:
* I made wipe_for_renegotiate and renegotiate share the same logic as wipe and negotiate. This means that we only have two well-define pub(crate) methods, AND that we won't accidentally forget to update the renegotiation logic when we update the main logic.
* I surfaced a new context_mut method for the config builder instead of piecemeal slapping pub(crate) on things. That'll also give us a clear path to other modules defining callback setters.

Reply:
Hmm, I do like it better, but I'd still prefer it to be part of the main connection/config implementation. But this is a two way door, so I'm also fine with merging this the way it is.

As for the `connection.rs` and friend being very large, I agree but we'll need to [fix that](https://gist.github.com/jmayclin/0ec2b4923904960b5d07b688af24f0ef#no-mega-modules) eventually anyways 😅.